Build and test with CUDA 13.0.0 (#1162)
rapidsai/ucx-py: docs/source/install.rst

::

>     # CUDA 13
>     conda create -n ucx -c conda-forge -c rapidsai \
>       cuda-version=13.0 ucx-py
> 
>     # CUDA 12
>     conda create -n ucx -c conda-forge -c rapidsai \
>       cuda-version=12.9 ucx-py

Reviewer: ```suggestion
    # CUDA 12
    conda create -n ucx -c conda-forge -c rapidsai \
      cuda-version=12.9 ucx-py

    # CUDA 13
    conda create -n ucx -c conda-forge -c rapidsai \
      cuda-version=13.0 ucx-py
```

Reviewer: Just to keep consistent ordering with text and keep chronological ordering.

Author: @bdice asked me in https://github.com/rapidsai/kvikio/pull/803#discussion_r2287989908 to order these types of things with newer CUDA first.

I'm indifferent about it, will let him comment here and will do whatever you two agree on.

Reviewer: I'd prefer we keep it consistent with the text, it looks very strange to me having 13 come before 12, this is how it always looked in the past (11 then 12).

Reviewer 2: We want to shift the default (first command offered) to be the newest. The newest CUDA version will be supported for longer and we want to encourage users to adopt new versions.

Reviewer 2: We can rearrange the text as needed but should prefer 13 over 12 in ordering and in any situations where we only give one example.

Reviewer: I'd prefer if we then rephrase (some of) the text, if we want to encourage that we could start referring to CUDA 12 as "legacy" or something like that to provide some strong encouragement.

Author: I don't think we should hold up this PR over this phrasing in documentation. Since you've both approved the PR (so approved all the other changes), I'm planning to merge this as soon as CI finishes, to keep moving forward with supporting CUDA 13 in RAPIDS.

I'd be happy to review a follow-up PR changing the text around these install instructions.

Reviewer: Yes, please move forward @jameslamb .

Author: Thank you!

PyPI
----

PyPI installation is possible and currently supports CUDA versions
``12`` and ``13``. Packages are compatible with CPU-only workloads and any one can
be chosen if the application doesn't use CUDA, but currently there are
no pre-built CPU-only packages available, so the CUDA package must be
installed instead. CUDA versions are differentiated by the suffix
``-cuXY``, where ``XY`` must be replaced with the desired CUDA version.

::

>     # CUDA 13
>     pip install ucx-py-cu13
> 
>     # CUDA 12
>     pip install ucx-py-cu12

Reviewer: ```suggestion
    # CUDA 12
    pip install ucx-py-cu12

    # CUDA 13
    pip install ucx-py-cu13
```

    # or for develop build
    pip install -v -e .

This will automatically handle installing appropriate, compatible ``libucx-cu{12,13}`` packages for build-time and runtime use.

> When you run UCX-Py code installed this way, it will load UCX libraries from the installed ``libucx-{cu12,13}`` package.

Reviewer 2: ```suggestion
When you run UCX-Py code installed this way, it will load UCX libraries from the installed ``libucx-cu{12,13}`` package.
```

Author: Oh totally agree! This was my intent, but clearly not what my fingers typed 🙃 

Thank you

UCX-Py packages are built against the oldest version of UCX that UCX-Py supports, and can run against a range
of ABI-compatible UCX versions.

You can use packages from PyPI to customize the UCX version used at runtime.
For example, to switch to using UCX 1.19 at runtime, run the following.

::

>     # CUDA 13
>     pip install 'libucx-cu13>=1.19.0,<1.20'
> 
>     # CUDA 12

Reviewer: ```suggestion
    # CUDA 12
    pip install 'libucx-cu12>=1.19.0,<1.20'

    # CUDA 13
```

>     pip install 'libucx-cu12>=1.19.0,<1.20'

Reviewer: ```suggestion
    pip install 'libucx-cu13>=1.19.0,<1.20'
```
